credentialProvider/src/main/java/nl/nn/credentialprovider/util/Cache.java

```diff
@@ -0,0 +1,41 @@
+/*
+   Copyright 2022 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.credentialprovider.util;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class Cache<K,V,E extends Exception> {
+
+	private Map<K,CacheEntry<V>> delegate;
+	private int timeToLiveMillis;
+
+	public Cache(int timeToLiveMillis) {
+		this.timeToLiveMillis = timeToLiveMillis;
+		delegate = new ConcurrentHashMap<>();
+	}
+
+	public V computeIfAbsentOrExpired(K key, ThrowingFunction<K,V,E> valueSupplier) throws E {
```

Comment:
Moet de hele method niet synchronized zijn?

Reply:
De concurrenthashmap levert altijd dezelfde entry voor dezelfde key, die is daarvoor gemaakt.
Wat er vervolgens met die entry gebeurt moet wel in een sync block. Ik heb de getValue() daar nu ook in getrokken.